spec/ddtrace/contrib/support/tracer_helpers.rb

```diff
@@ -71,10 +71,17 @@ def clear_spans!
           instance
         end
       end
+
+      config.around(:each) do |example|
+        example.run.tap do
+          Datadog.tracer.shutdown!
+        end
+      end
```

Comment:
This was originally done to ensure that the tracer shutdown is run before any `after` blocks, but I don't recall if this is necessary anymore at this point. I'll try removing it.

Reply:
Turns out this was the secret to all things: executing it at `around` allow us to execute code outside of the example execution. This allows us to run code _after_ all test code (including mock verification and tear down) have been run. This is very important here, as we don't want `Datadog.tracer.shutdown!` tripping test assertions.

Also, this guarantees we are running after all teardown is complete.

So I tried changing this to `after(:each`, all hell broke loose, so I reverted it. But it turns out I can use `around(:each)` for the main background thread verification. It's currently done at `config.after(:all)` in the spec_helper. I'm changing it to `around` now, as it will more closely be able to report on the exact test that leaked. 🤞 it works well.

Reply:
Noice! 🥇